oratorio/coach/models.py

```diff
@@ -16,6 +16,48 @@ class User(models.Model):
     def __str__(self):
        return "User " + self.name + ", email " + self.email
 
+    def get_avg_pace(self):
+        "Return the average pace of all the speeches this user owns"
+        res = 0
+        speeches = Speech.objects.filter(user=self)
+        if speeches:
+            for speech in speeches:
+                res += speech.get_avg_pace()
+            res /= len(speeches)
+        return res
+    
+    def get_avg_tone(self):
+        "Return a dict where keys are types of emotions and values are the \
+        average values of those emotions for the user"
+        speeches = Speech.objects.filter(user=self)
+        res = { 'joy': 0, 'sadness': 0, 'anger': 0,
+                'fear': 0, 'disgust': 0, 'confident': 0 }
+        if speeches:
+            for speech in speeches:
+                avg_tone = speech.get_avg_tone()
+                res['joy'] += avg_tone['joy']
+                res['sadness'] += avg_tone['sadness']
+                res['anger'] += avg_tone['anger']
+                res['fear'] += avg_tone['fear']
+                res['disgust'] += avg_tone['disgust']
+                res['confident'] += avg_tone['confident']
+            res['joy'] /= len(speeches)
```

Comment:
Error divide by zero. Although this is python, which appears to take care of this sort of issue? It didn't crash when I ran it....